colmena/redis/queue.py

```diff
@@ -234,39 +234,75 @@ def __init__(self, hostname: str, port: int = 6379, name: Optional[str] = None,
             serialization_method (SerializationMethod): Method used to store the input
             keep_inputs (bool): Whether to keep inputs after method results are stored
             topics ([str]): List of topics used when having the client filter different types of tasks
-            value_server_threshold (int): Threshold (bytes) to store objects in
-                value server. If None, the value server is not used.
-            value_server_hostname (str): Optional redis server hostname for the
-                value server. If not provided and the value server is used,
-                defaults to `hostname`.
-            value_server_port (str): Optional redis server port for the
-                value server. If not provided and the value server is used,
-                defaults to `port`.
+            proxystore_name (str, dict): Name of ProxyStore backend to use for all
+                topics or a mapping of topic to ProxyStore backend for specifying
+                backends for certain tasks. If a mapping is provided but a topic is
+                not in the mapping, ProxyStore will not be used.
+            proxystore_threshold (int, dict): Threshold in bytes for using
+                ProxyStore to transfer objects. Optionally can pass a dict
+                mapping topics to threshold to use different threshold values
+                for different topics. None values in the mapping will exclude
+                ProxyStore use with that topic.
         """
 
         # Store the result communication options
         self.serialization_method = serialization_method
         self.keep_inputs = keep_inputs
-        self.value_server_threshold = value_server_threshold
-
-        if self.value_server_threshold is not None:
-            if self.serialization_method.lower() != 'pickle':
-                raise ValueError('Serialization method must be pickle to use the value server')
-            if value_server_hostname is None:
-                value_server_hostname = hostname
-            if value_server_port is None:
-                value_server_port = port
-            ps.store.init_store(
-                ps.store.STORES.REDIS,
-                name='redis',
-                hostname=value_server_hostname,
-                port=value_server_port
-            )
-            logger.debug('Initialized value server using Redis server at '
-                         f'{value_server_hostname}:{value_server_port}')
-
-        self.value_server_hostname = value_server_hostname
-        self.value_server_port = value_server_port
+
+        # Create {topic: proxystore_name} mapping
+        if isinstance(proxystore_name, str):
+            self.proxystore_name = defaultdict(lambda: proxystore_name)
+        elif isinstance(proxystore_name, dict):
+            self.proxystore_name = defaultdict(lambda: None)
+            for topic, ps_name in proxystore_name:
```

Comment:
Could you use `dict.update` for this instead?

Reply:
Yep! That's a better method. 

My first though was to use dictionary expansion `{**a, **b}` to merge but interestingly that casts defaultdicts to dicts.